### PR TITLE
Feature: migrate api

### PR DESCRIPTION
**Pull request by moiskillnadne**

### CODE REVIEW

File: src/features/ChangePassword/ui/ChangePasswordForm.tsx [hunk 1]
```diff
@@ -32,26 +34,41 @@ export const ChangePasswordForm = ({ title, token, temporaryToken, onSuccessExte
...
 
-    if (token && !temporaryToken && data.old) {
-      return updatePassword({ token, new: data.new, old: data.old })
+    if (!token && data.old) {
+      return updatePassword({ password: data.new, oldPassword: data.old })
     }
   }
 
+  const approveErrorMessage = useMemo(() => approveError?.response?.data?.messages[0], [approveError])
+  const updateErrorMessage = useMemo(() => updateError?.response?.data?.messages[0], [updateError])
```
Comment (BamMironov) [inline src/features/ChangePassword/ui/ChangePasswordForm.tsx:62]: Please create a shared function called "selectFirstError" or something. 

`const selectFirstError = error => error?.response?.data?.messages[0]`

then create a shared hook called useFirstErrorMessage:
```
const useFirstErrorMessage = (error: BaseError | null) => {
  return useMemo(() => selectFirstError(error), [error]);
}
```

File: src/features/ForgotPassword/ui/ForgotPasswordForm.tsx [hunk 2]
```diff
@@ -40,7 +40,7 @@ export const ForgotPasswordForm = () => {
           </BasicButton>
         )}
 
-        {isSuccess && <DisplaySystemMessage successMessage={data?.data?.message} />}
+        {/* {isSuccess && <DisplaySystemMessage successMessage={data?.data?.result?.message} />} */}
```
Comment (BamMironov) [inline src/features/ForgotPassword/ui/ForgotPasswordForm.tsx:43]: Why did you comment this?

File: src/shared/api/hooks/useBaseMutation.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  options?: Omit<
+    UseMutationOptions<AxiosResponse<TResponse, BaseError>, BaseError, TRequest>,
+    "mutationKey" | "mutationFn"
+  >,
+) => {
+  return useMutation(key, mutationFn, {
+    ...options,
+    onError: (error: BaseError, variables: TRequest, context: unknown) => {
+      if (error?.response?.data?.messages?.length) {
+        error.evaluatedMessage = error.response?.data?.messages[0]
```
Comment (BamMironov) [inline src/shared/api/hooks/useBaseMutation.ts:18]: Well, if you have this one then you can make use of it instead of creating useFirstErrorMessage

**Review by BamMironov — CHANGES_REQUESTED** [review]

File: src/features/ChangePassword/ui/ChangePasswordForm.tsx (hunk 1 above)
Comment (moiskillnadne) [reply src/features/ChangePassword/ui/ChangePasswordForm.tsx:62]: Done

File: src/shared/api/hooks/useBaseMutation.ts (hunk 3 above)
Comment (moiskillnadne) [reply src/shared/api/hooks/useBaseMutation.ts:18]: Done

File: src/features/ForgotPassword/ui/ForgotPasswordForm.tsx [hunk 4]
```diff
@@ -9,6 +11,7 @@ import { useCustomForm } from "~/shared/utils"
 
 export const ForgotPasswordForm = () => {
   const { t } = useForgotPasswordTranslation()
+  const [email, setEmail] = useState("")
```
Comment (BamMironov) [inline src/features/ForgotPassword/ui/ForgotPasswordForm.tsx:14]: Please use watch api instead https://react-hook-form.com/api/useform/watch

**Review by BamMironov — CHANGES_REQUESTED** [review]

File: src/features/ForgotPassword/ui/ForgotPasswordForm.tsx (hunk 4 above)
Comment (moiskillnadne) [reply src/features/ForgotPassword/ui/ForgotPasswordForm.tsx]: Done

**Review by BamMironov — APPROVED** [review]

**Pull request merged**